test/integration/confirmations/signatures/personalSign.test.tsx

```diff
@@ -0,0 +1,190 @@
+import { fireEvent, waitFor } from '@testing-library/react';
+import { ApprovalType } from '@metamask/controller-utils';
+import mockMetaMaskState from '../../data/integration-init-state.json';
+import { integrationTestRender } from '../../../lib/render-helpers';
+import { shortenAddress } from '../../../../ui/helpers/utils/util';
+import * as backgroundConnection from '../../../../ui/store/background-connection';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+  MetaMetricsEventLocation,
+} from '../../../../shared/constants/metametrics';
+import { MESSAGE_TYPE } from '../../../../shared/constants/app';
+
+jest.mock('../../../../ui/store/background-connection', () => ({
+  ...jest.requireActual('../../../../ui/store/background-connection'),
+  submitRequestToBackground: jest.fn(),
+}));
+
+const mockedBakcgroundConnection = jest.mocked(backgroundConnection);
+
+const backgroundConnectionMocked = {
+  onNotification: jest.fn(),
+};
+
+const getMetaMaskStateWithUnapprovedPersonalSign = (accountAddress: string) => {
+  const pendingPersonalSignId = '0050d5b0-c023-11ee-a0cb-3390a510a0ab';
+  const pendingPersonalSignTime = new Date().getTime();
+  return {
+    ...mockMetaMaskState,
+    preferences: {
+      ...mockMetaMaskState.preferences,
+      redesignedConfirmationsEnabled: true,
+    },
+    unapprovedPersonalMsgs: {
+      [pendingPersonalSignId]: {
+        id: pendingPersonalSignId,
+        status: 'unapproved',
+        time: pendingPersonalSignTime,
+        type: MESSAGE_TYPE.PERSONAL_SIGN,
+        securityProviderResponse: null,
+        msgParams: {
+          from: accountAddress,
+          data: '0x4578616d706c652060706572736f6e616c5f7369676e60206d657373616765',
+          origin: 'https://metamask.github.io',
+          siwe: { isSIWEMessage: false, parsedMessage: null },
+        },
+      },
+    },
+    unapprovedPersonalMsgCount: 1,
+    pendingApprovals: {
+      [pendingPersonalSignId]: {
+        id: pendingPersonalSignId,
+        origin: 'origin',
+        time: pendingPersonalSignTime,
+        type: ApprovalType.PersonalSign,
+        requestData: {},
+        requestState: null,
+        expectsResult: false,
+      },
+    },
+    pendingApprovalCount: 1,
+  };
+};
+
+describe('PersonalSign Confirmation', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('displays the header account modal with correct data', async () => {
+    const account =
+      mockMetaMaskState.internalAccounts.accounts[
+        mockMetaMaskState.internalAccounts
+          .selectedAccount as keyof typeof mockMetaMaskState.internalAccounts.accounts
```

Comment:
Hmm, it looks like the type error you're getting past with this cast is caused by TypeScript interpreting the initial state as a constant. Ideally we'd widen that type to reflect the actual MetaMask state shape, but we don't have a type for that yet. Hopefully this is something we can return to soon.